modules/passkey/contracts/4337/SafeECDSASignerLaunchpad.sol

```diff
@@ -243,37 +279,77 @@ contract SafeECDSASignerLaunchpad is IAccount, SafeStorage {
         _setInitHash(0);
     }
 
-    function _domainSeparator() internal view returns (bytes32) {
+    /**
+     * @notice Computes the EIP-712 domain separator for Safe launchpad operations.
+     * @return domainSeparatorHash The EIP-712 domain separator hash for this contract.
+     */
+    function domainSeparator() public view returns (bytes32) {
         return keccak256(abi.encode(DOMAIN_SEPARATOR_TYPEHASH, block.chainid, SELF));
     }
 
-    function _getOperationData(
+    /**
+     * @dev Validates that the user operation is correctly signed and returns an ERC-4337 packed validation data
+     * of `validAfter || validUntil || authorizer`:
+     *  - `authorizer`: 20-byte address, 0 for valid signature or 1 to mark signature failure (this module does not make use of signature aggregators).
+     *  - `validUntil`: 6-byte timestamp value, or zero for "infinite". The user operation is valid only up to this time.
+     *  - `validAfter`: 6-byte timestamp. The user operation is valid only after this time.
+     * @param userOp User operation struct.
+     * @return validationData An integer indicating the result of the validation.
+     */
+    function _validateSignatures(
+        PackedUserOperation calldata userOp,
         bytes32 userOpHash,
-        uint48 validAfter,
-        uint48 validUntil
-    ) internal view returns (bytes memory operationData) {
-        operationData = abi.encodePacked(
-            bytes1(0x19),
-            bytes1(0x01),
-            _domainSeparator(),
-            keccak256(abi.encode(SAFE_INIT_OP_TYPEHASH, userOpHash, validAfter, validUntil, SUPPORTED_ENTRYPOINT))
-        );
+        address signerFactory,
+        uint256 signerX,
+        uint256 signerY,
+        address signerVerifier
+    ) internal view returns (uint256 validationData) {
+        uint48 validAfter;
+        uint48 validUntil;
+        bytes calldata signature;
+        {
+            bytes calldata sig = userOp.signature;
+            validAfter = uint48(bytes6(sig[0:6]));
+            validUntil = uint48(bytes6(sig[6:12]));
+            signature = sig[12:];
+        }
+
+        bytes32 operationHash = getOperationHash(userOpHash, validAfter, validUntil);
+        try
+            ICustomECDSASignerFactory(signerFactory).isValidSignatureForSigner(operationHash, signature, signerX, signerY, signerVerifier)
+        returns (bytes4 magicValue) {
+            // The timestamps are validated by the entry point, therefore we will not check them again
+            validationData = _packValidationData(magicValue != ERC1271.MAGIC_VALUE, validUntil, validAfter);
+        } catch {
+            validationData = _packValidationData(true, validUntil, validAfter);
+        }
     }
 
+    /**
+     * @notice Reads the configured initialization hash from storage.
+     */
```

Comment:
nit: should this specify the return value type? similar question about `isContract`

Reply:
82ba1ef19f95fbbd8a382fc96a08b611ba27bc08